.github/workflows/contracts-deploy.yml

```diff
@@ -46,7 +48,14 @@ jobs:
              - '.github/workflows/contracts-deploy.yml'
 
   build-and-deploy-contracts:
-    needs: [detect-contracts-changes]
+    # TODO: e2e test are flaky AF
+    # and they block the deployment
+    # which in turns stalls development
+    # as frontend developers rely heavily on the devnet environment.
+    # Therefore we made an executive decision to deploy to the devnet
+    # regardless of the contracts e2e tests status.
+    # Once fixed this dependency can be brough back
+    # needs: [detect-contracts-changes]
```

Comment:
I think it's a compromise of approach of using paths and not using paths to detect if pull request should trigger. In this solution, this PR would trigger always in GH interface, yet it will be run actually only on those path changes. If we ever go back to this solution, this is the best option

Reply:
exactly :point_up: